compute_score.py

```diff
@@ -7,63 +7,51 @@
     parser.add_argument("--benchmark_data_dir",
         help="directory containing multiple .json files for each of which to compute a score")
-    parser.add_argument('--hack_data', nargs=2, action='append', help="keyword to match benchmark names, and multiplicative factor to adjust their measurement")
     args = parser.parse_args()
 
-    config = None
-    if args.configuration is not None:
-        with open(args.configuration) as cfg_file:
-            config = yaml.full_load(cfg_file)
-    else:
-        if args.benchmark_data_file is None and args.benchmark_data_dir is None:
-            parser.print_help(sys.stderr)
-            raise ValueError("Invalid command-line arguments. You must specify a config, a data file, or a data dir.")
-        with open(SPEC_FILE_DEFAULT) as spec_file:
-            spec = yaml.full_load(spec_file)
+    if args.benchmark_data_file is None and args.benchmark_data_dir is None:
+        parser.print_help(sys.stderr)
+        raise ValueError("Invalid command-line arguments. You must specify a config, a data file, or a data dir.")
 
+    files = []
+    benchmark_data = []
+    scores = []
     if args.benchmark_data_file is not None:
         with open(args.benchmark_data_file) as data_file:
             data = json.load(data_file)
-            if config is None:
-                config = generate_bench_cfg(spec, data, TARGET_SCORE_DEFAULT)
-        score = compute_score(config, data)
-        print(score)
-        if args.hack_data:
-            fake_data = {}
-            for keyword, factor in args.hack_data:
-                for b in data['benchmarks']:
-                    if keyword.lower() in b['name'].lower():
-                        fake_data[b['name']] =  b['stats']['mean'] * float(factor)
-
-            hacked_score = compute_score(config, data, fake_data)
-            print(f"Using hacks {args.hack_data}, hacked_score {hacked_score}")
-
+            benchmark_data.append(data)
     elif args.benchmark_data_dir is not None:
-        scores = [('File', 'Score', 'PyTorch Version')]
         for f in os.listdir(args.benchmark_data_dir):
             path = os.path.join(args.benchmark_data_dir, f)
             if os.path.isfile(path) and os.path.splitext(path)[1] == '.json':
                 with open(path) as data_file:
                     data = json.load(data_file)
-                    if config is None:
-                        config = generate_bench_cfg(spec, data, TARGET_SCORE_DEFAULT)
-                    score = compute_score(config, data)
-                    scores.append((f, score, data['machine_info']['pytorch_version']))
+                    files.append(f)
+                    benchmark_data.append(data)
+
+    if args.configuration == "relative":
+        ref_data = benchmark_data[0]
+        score_config = TorchBenchScore(ref_data)
+    elif args.configuration == "v0":
+        score_config = TORCHBENCH_V0_SCORE
+    else:
+        raise ValueError("Invalid score configuration")
```

Comment:
Is it a problem if the reference json and the SPEC_FILE_DEFAULT doesn't match? E.g. only part of the benchmarks in SPEC_FILE_DEFAULT is in reference json.

Reply:
So only those benchmarks which are available in the reference json, will be accounted for scoring. So it should not be a problem if the ref json contains only part of the SPEC_FILE_DEFAULT.

Reply:
Makes sense!